spec/facter/facts/solaris/is_virtual_spec.rb

```diff
@@ -25,16 +26,33 @@
         allow(Facter::Resolvers::Solaris::Dmi).to receive(:resolve).with(:product_name).and_return('unkown')
         allow(Facter::Resolvers::Solaris::Dmi).to receive(:resolve).with(:bios_vendor).and_return('unkown')
         allow(Facter::Resolvers::VirtWhat).to receive(:resolve).with(:vm).and_return(nil)
+        allow(Facter::Resolvers::Xen).to receive(:resolve).with(:vm).and_return(nil)
       end
 
-      it 'returns is_virtual fact as physical' do
+      it 'returns is_virtual fact as false' do
         expect(fact.call_the_resolver).to be_an_instance_of(Facter::ResolvedFact).and \
           have_attributes(name: 'is_virtual', value: vm)
       end
     end
 
-    context 'when ldom hypervisor is found' do
+    context 'when Ldom role_control is false, ldom hypervisor is found' do
       let(:vm) { true }
+      let(:role_control) { 'false' }
+
+      before do
+        allow(Facter::Resolvers::Solaris::Ldom).to receive(:resolve).with(:role_impl).and_return(vm)
+        allow(Facter::Resolvers::Solaris::Ldom).to receive(:resolve).with(:role_control).and_return(role_control)
+      end
+
+      it 'returns is_virtual fact as true' do
+        expect(fact.call_the_resolver).to be_an_instance_of(Facter::ResolvedFact).and \
+          have_attributes(name: 'is_virtual', value: vm)
```

Comment:
Vm should be `LDoms` in this case.

Reply:
is_virtual returns true or false. virtual may return LDoms